### ISSUE TITLE
Search one partition results wrong after creating index 

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#6713]
#### Steps/Code to reproduce:
/milvus/tests20/python_client/testcases/test_search.py::TestCollectionSearch::test_search_index_one_partition

#### Expected result:
Search results are equal with limit

#### Actual results:
```
FAILED test_search.py::TestCollectionSearch::test_search_index_one_partition[5000-False-False] - AssertionError
FAILED test_search.py::TestCollectionSearch::test_search_index_one_partition[5000-False-True] - AssertionError
FAILED test_search.py::TestCollectionSearch::test_search_index_one_partition[5000-True-True] - AssertionError

[2021-07-21 12:03:32 - ERROR - ci_test]: search_results_check: limit(topK) searched (993) is not equal with expected (1000) (func_check.py:186)
[2021-07-21 12:03:41 - ERROR - ci_test]: search_results_check: limit(topK) searched (765) is not equal with expected (1000) (func_check.py:186)
[2021-07-21 12:03:59 - ERROR - ci_test]: search_results_check: limit(topK) searched (906) is not equal with expected (1000) (func_check.py:186)
```

#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93): latest
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2):
Name: pymilvus
Version: 2.0.0rc3.dev6
Name: pymilvus-orm
Version: 2.0.0rc3.dev6

- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:

**Comment by DragonDriver** [comment]
I will fix this soon. thx for pointing out this.

**Comment by DragonDriver** [comment]
Reproduced on my local pc. But if i search on a collection without creating index (create collection, insert data, flush, load), the number of hit records (topk) is equal to limit. So it's `IVF_FLAT` caused this issue. `nprobe` (10) in test cases is too small here. When we create index, we set `nlist` to 128, `IVF_FALT` will divide vectors to 128 groups. Now suppose `IVF_FLAT` divide insert dataset evenly, the number of vectors per group is even less than 40, so the `nprobe`(10) * 40 = 400 is also less than needed limit. Certainly, `IVF_FLAT` will divide vectors by the characteristic of dataset, not evenly, so some groups will be more sparser. It means less records will be searched. Any way, `nprobe` here is too small.
